src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -170,5 +172,40 @@ private void AddRequestHeadersAndSetRequestUri(HttpRequestMessage httpRequestMes
             httpRequestMessage.Headers.Add(HttpWorkerConstants.HostVersionHeaderName, ScriptHost.Version);
             httpRequestMessage.Headers.UserAgent.ParseAdd($"{HttpWorkerConstants.UserAgentHeaderValue}/{ScriptHost.Version}");
         }
+
+        public async Task<bool> IsWorkerReady()
+        {
+            await Utility.DelayAsync(WorkerConstants.WorkerInitTimeoutSeconds, WorkerConstants.PollingIntervalMilliseconds, async () =>
+            {
+                return await ShouldContinueWaitingForWorker();
+            });
+            bool continueWaitingForWorker = await ShouldContinueWaitingForWorker();
```

Comment:
A bit unfortunate to have this additional call here as it won't be necessary in most cases and it is an expense we can avoid. If the delay was modified to return a `Task<bool>` you could just inspect that result instead.